### ISSUE TITLE
shellcheck source= sometimes conflict with regular source calls, leading to spurious SC2154

### ISSUE THREAD

**Issue by knirch** [koalaman/shellcheck#1692]
a
```
FOO=1
foo=1
```
b
```
BAR=1
bar=1
```

ex1
```
#!/bin/bash
source a
source b
echo "$FOO $BAR"
echo "$foo $bar"
```
ex2
```
#!/bin/bash
# shellcheck source=a
source /dev/null
source b
echo "$FOO $BAR"
echo "$foo $bar"
```
ex3
```
#!/bin/bash
_=1
# shellcheck source=a
source /dev/null
source b
echo "$FOO $BAR"
echo "$foo $bar"
```
ex4
```
#!/bin/bash
source b
# shellcheck source=a
source /dev/null
echo "$FOO $BAR"
echo "$foo $bar"
```
ex5
```
#!/bin/bash
# shellcheck source=a
source /dev/null
_=1
source b
echo "$FOO $BAR"
echo "$foo $bar"
```

Result:
```
knirch@traktor:~/shellcheck-test-case$ shellcheck -o check-unassigned-uppercase -x ex*
ex2:5:12: warning: BAR is referenced but not assigned. [SC2154]
ex2:6:12: warning: bar is referenced but not assigned. [SC2154]
ex5:6:12: warning: BAR is referenced but not assigned. [SC2154]
ex5:7:12: warning: bar is referenced but not assigned. [SC2154]
```

I find it particularly interesting that ex3 doesn't trigger the behaviour.

Happy hunting :)
